src/node_nanomsg.cc

```diff
@@ -199,21 +199,24 @@ NAN_METHOD(Shutdown) {
 NAN_METHOD(Send) {
   NanScope();
 
-  int s = args[0]->Uint32Value();
+  int flags = args[2].integer;
+  std::string *input;
 
-  if (!node::Buffer::HasInstance(args[1])) {
-    return NanThrowError("second Argument must be a Buffer.");
-  }
+  if (node::Buffer::HasInstance(args[1]->ToObject())) {
 
-  Local<Object> obj = args[1]->ToObject();
-  char *odata = node::Buffer::Data(obj);
-  size_t odata_len = node::Buffer::Length(obj);
-  int flags = args[2]->Uint32Value();
+    v8::Handle<v8::Object> object = args[1]->ToObject();
 
-  // Invoke nanomsg function.
-  int ret = nn_send(s, odata, odata_len, flags);
+    const char *data = node::Buffer::Data(object);
+    input = new std::string(data, node::Buffer::Length(object));
```

Comment:
Doesn't std::string free the allocation? input gets deleted by the C++ class. And NanNew creates a temporary reference that gets recycled when nanscope is closed, so that gets taken care of.